### ISSUE TITLE
exporters/stats/stackdriver: warnings about no return for missing default case

### ISSUE THREAD

**Issue by odeke-em** [census-instrumentation/opencensus-cpp#162]
In this code below
https://github.com/census-instrumentation/opencensus-cpp/blob/cd3f7109e5ac046c2c0347c68eeeb7f544714072/opencensus/exporters/stats/stackdriver/internal/stackdriver_utils.cc#L189-L200

we can see that we have 3 different cases that all have returns but the default case doesn't

Compiling code with it with `-Wreturn-type` gives a bunch of warnings
```shell
INFO: From Compiling opencensus/exporters/stats/stackdriver/internal/stackdriver_utils.cc:
opencensus/exporters/stats/stackdriver/internal/stackdriver_utils.cc: In function 'google::api::MetricDescriptor::ValueType opencensus::exporters::stats::{anonymous}::GetValueType(const opencensus::stats::ViewDescriptor&)':
opencensus/exporters/stats/stackdriver/internal/stackdriver_utils.cc:86:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
opencensus/exporters/stats/stackdriver/internal/stackdriver_utils.cc: In function 'std::vector<google::monitoring::v3::TimeSeries> opencensus::exporters::stats::MakeTimeSeries(const opencensus::stats::ViewDescriptor&, const opencensus::stats::ViewData&, absl::string_view)':
opencensus/exporters/stats/stackdriver/internal/stackdriver_utils.cc:200:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
```
and in other places in the code.

In other languages such as Go that code wouldn't compile

**Comment by isturdy** [comment]
Duplicate of #79--we do not provide a fallthrough intentionally, as these cases should never be reached barring memory corruption (or someone explicitly casting ints to enums, which is not supported), and we do not see a way of handling these without cluttering the generated code.

**Comment by g-easy** [comment]
Fixed as part of #79.
